lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java

```diff
@@ -95,15 +95,16 @@
 
     this.handlerContainer.setTransport(this.transport);
 
-    this.myIdentifier = new SocketRemoteIdentifier((InetSocketAddress)this.transport.getLocalAddress());
+    InetSocketAddress address = new InetSocketAddress(localAddressProvider.getLocalAddress(), this.transport.getListeningPort());
```

Comment:
This seems not quite right to me.  Shouldn't we keep the original code here but change the address, port in Transport instance?

Reply:
The transport instance is how the container communicates externally which is already driven by the inject to the NettyMessagingTransport class. We have this line to be able to determine the right value of this.myIdentifier.

---
In reply to: [196590120](https://github.com/sharathmalladi/reef/pull/5#discussion_r196590120) [](ancestors = 196590120)